caikit_nlp/modules/embedding_retrieval/embedding.py

```diff
@@ -0,0 +1,140 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Standard
+from pathlib import Path
+from typing import List
+import os
+
+# Third Party
+from sentence_transformers import SentenceTransformer
+
+# First Party
+from caikit.core import ModuleBase, ModuleConfig, ModuleSaver, module
+from caikit.core.exceptions import error_handler
+import alog
+
+# Local
+from .embedding_retrieval_task import EmbeddingRetrievalTask
+from caikit_nlp.data_model.embedding_vectors import EmbeddingResult, Vector1D
+
+logger = alog.use_channel("<EMBD_BLK>")
+error = error_handler.get(logger)
+
+HOME = Path.home()
+DEFAULT_HF_MODEL = "sentence-transformers/all-MiniLM-L6-v2"
```

Comment:
Modules usually doesn't work with default model paradigm, what is the particular use-case for this?

Reply:
This is a convenience (think demo), but I think that doesn't apply to caikit-nlp.  I'll make this fail if the config doesn't have model.

Reply:
Ideally, we would like to make the specification of list of models (hugging face or local) part of the deployment time parameters rather than hard-coded in the service.